kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManager.java

```diff
@@ -91,16 +91,21 @@ private static ExecutorService queryThreadPool() {
     // Using the spotify library ( this method is much easier to operate then using
     // CompletableFuture.allOf and converting the CompletableFuture<Void> to
     // CompletableFuture<List<SearchResult>>
-    CompletableFuture<List<SearchResult<T>>> searchResults = CompletableFutures.allAsList(queries);
+    CompletableFuture<List<SearchResult<T>>> searchResultFuture =
+        CompletableFutures.allAsList(queries);
     try {
+      List<SearchResult<T>> searchResults =
+          searchResultFuture.get(LOCAL_QUERY_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
       //noinspection unchecked
       SearchResult<T> aggregatedResults =
           ((SearchResultAggregator<T>) new SearchResultAggregatorImpl<>(query))
-              .aggregate(searchResults.get(30, TimeUnit.SECONDS));
+              .aggregate(searchResults);
       return incrementNodeCount(aggregatedResults);
     } catch (Exception e) {
       LOG.error("Error searching across chunks ", e);
       throw new RuntimeException(e);
+    } finally {
```

Comment:
Please add a comment on why we should cancel this future here so in future, we don't delete the line by mistake.

Reply:
Done - both here and on the other future cancellation section.